crates/lsp/src/handlers.rs

```diff
@@ -24,30 +26,55 @@ pub(crate) async fn handle_initialized(
     let span = tracing::info_span!("handle_initialized").entered();
 
     // Register capabilities to the client
-    let mut regs: Vec<lsp_types::Registration> = vec![];
+    let mut registrations: Vec<lsp_types::Registration> = vec![];
 
-    if lsp_state.needs_registration.did_change_configuration {
+    if lsp_state
+        .capabilities
+        .dynamic_registration_for_did_change_configuration
+    {
         // The `didChangeConfiguration` request instructs the client to send
         // a notification when the tracked settings have changed.
         //
         // Note that some settings, such as editor indentation properties, may be
         // changed by extensions or by the user without changing the actual
         // underlying setting. Unfortunately we don't receive updates in that case.
-        let mut config_document_regs = collect_regs(
+        let mut config_document_registrations = collect_regs(
             VscDocumentConfig::FIELD_NAMES_AS_ARRAY.to_vec(),
             VscDocumentConfig::section_from_key,
         );
-        let mut config_diagnostics_regs: Vec<lsp_types::Registration> = collect_regs(
+        let mut config_diagnostics_registrations: Vec<lsp_types::Registration> = collect_regs(
             VscDiagnosticsConfig::FIELD_NAMES_AS_ARRAY.to_vec(),
             VscDiagnosticsConfig::section_from_key,
         );
 
-        regs.append(&mut config_document_regs);
-        regs.append(&mut config_diagnostics_regs);
+        registrations.append(&mut config_document_registrations);
+        registrations.append(&mut config_diagnostics_registrations);
+    }
+
+    if lsp_state
+        .capabilities
+        .dynamic_registration_for_did_change_watched_files
+    {
+        // Watch for changes in `air.toml` files so we can react dynamically
+        let watch_air_toml_registration = lsp_types::Registration {
+            id: uuid::Uuid::new_v4().to_string(),
+            method: "workspace/didChangeWatchedFiles".into(),
+            register_options: Some(
+                serde_json::to_value(DidChangeWatchedFilesRegistrationOptions {
+                    watchers: vec![FileSystemWatcher {
+                        glob_pattern: lsp_types::GlobPattern::String("**/air.toml".into()),
+                        kind: None,
+                    }],
+                })
+                .unwrap(),
+            ),
+        };
+
+        registrations.push(watch_air_toml_registration);
```

Comment:
Watching for changes in `air.toml` files anywhere within a workspace (creation, deletion, modify)